cpp/tests/test_binaryop.cpp

```diff
@@ -14,86 +14,164 @@
  * limitations under the License.
  */
 
-#include <legate.h>
-
+#include <arrow/api.h>
+#include <arrow/compute/api.h>
 #include <cudf/binaryop.hpp>
-#include <cudf_test/base_fixture.hpp>
-#include <cudf_test/column_utilities.hpp>
-#include <cudf_test/column_wrapper.hpp>
-#include <cudf_test/type_lists.hpp>
+#include <gtest/gtest.h>
+#include <legate.h>
 
+#include "test_utils.hpp"
 #include <legate_dataframe/binaryop.hpp>
 #include <legate_dataframe/core/column.hpp>
 #include <legate_dataframe/core/table.hpp>
 
 using namespace legate::dataframe;
 
 template <typename T>
-struct BinaryOpsTest : public cudf::test::BaseFixture {};
+struct BinaryOpsTest : public testing::Test {};
 
-TYPED_TEST_SUITE(BinaryOpsTest, cudf::test::NumericTypes);
+// Edge cases for unsupported operations or inconsistent implementations between cudf/arrow
+bool skip(std::string const& op, legate::Type const& type)
+{
+  bool use_cudf =
+    legate::Runtime::get_runtime()->get_machine().count(legate::mapping::TaskTarget::GPU) > 0;
+  if (use_cudf) {
+    // Cudf integer power appears broken
+    // https://github.com/rapidsai/cudf/issues/10178#issuecomment-3004143727
+    if (op == "power" && type.to_string().find("int") != std::string::npos) { return true; }
```

Comment:
```suggestion
    if (op == "power" && cudf::is_integral(type)) { return true; }
```
with `#include <cudf/utilities/traits.hpp>`